utils/mail.go

```diff
@@ -100,14 +103,22 @@ func connectToSMTPServer(config *model.Config) (net.Conn, *model.AppError) {
 	return conn, nil
 }
 
-func newSMTPClient(conn net.Conn, config *model.Config) (*smtp.Client, *model.AppError) {
-	c, err := smtp.NewClient(conn, config.EmailSettings.SMTPServer+":"+config.EmailSettings.SMTPPort)
+func ConnectToSMTPServer(config *model.Config) (net.Conn, *model.AppError) {
+	return ConnectToSMTPServerAdvanced(
+		config.EmailSettings.ConnectionSecurity,
+		*config.EmailSettings.SkipServerCertificateVerification,
+		config.EmailSettings.SMTPServer,
+		config.EmailSettings.SMTPPort,
+	)
+}
+
+func NewSMTPClientAdvanced(conn net.Conn, connectionSecurity string, skipCertVerification bool, smtpServer string, smtpPort string, hostname string, auth bool, smtpUsername string, smtpPassword string) (*smtp.Client, *model.AppError) {
```

Comment:
This function takes nine parameters but could be decreased to four if SMTP values were in a `struct`.

Reply:
Done! :)